numpy/lib/npyio.py

```diff
@@ -427,10 +426,10 @@ def load(file, mmap_mode=None, allow_pickle=False, fix_imports=True,
         fid.seek(-min(N, len(magic)), 1)  # back-up
         if magic.startswith(_ZIP_PREFIX) or magic.startswith(_ZIP_SUFFIX):
             # zip-file (assume .npz)
-            # Transfer file ownership to NpzFile
+            # Potentially transfer file ownership to NpzFile
+            stack.pop_all()
             ret = NpzFile(fid, own_fid=own_fid, allow_pickle=allow_pickle,
                           pickle_kwargs=pickle_kwargs)
```

Comment:
Ideally we could pass the result of `stack.pop_all` directly into `NpzFile`, but perhaps best left to a follow-up PR

Reply:
ideally yes, but I, maybe someone else does, don't see how to do that without changing NpzFile's signature.

Reply:
Let's not worry about it here